### DISCUSSION TITLE
openai-whisper does not use GPU or CUDA

### DISCUSSION THREAD

**Discussion by audioscavenger** [openai/whisper#1694] [General]
just installed it with `pip install -U openai-whisper` on an Windows environment that has nVidia and where SD and h2oai work perfectly.

openai-whisper never uses the GPU and errors out when adding `--device cuda`:
`RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.`
How do we activate CUDA? do we need to compile it from source? it's not in the README. where is the wiki for Windows?

cuda version: `NVIDIA-SMI 528.24       Driver Version: 528.24       CUDA Version: 12.0`

**Comment by audioscavenger** [comment]
let's close this. now using whisper-faster since you do not offer install guide for cuda. appreciate the project tho. thanks

**Comment by qslug** [comment]
In case anyone else stumbles across this, make sure you install torch by passing in the CUDA version, not just `pip install torch`.

Per [PyTorch Start Locally](https://pytorch.org/get-started/locally/) I used:
```sh
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121
```

**Comment by pangyulei** [comment]
according to [PyTorch Official](https://pytorch.org/get-started/locally/) PyTorch only support python 3.8-3.11 currently, use `python -V` in CMD to check your python version, if not suitable, you have to reinstall python first. In Windows OS you do this:
1. uninstall python software (through control panel or other manage software you like)
2. remove python's path in environment variables
3. download python 3.11 exe from python official site, and install it
4. in CMD use `python -V` check again

in [PyTorch Official](https://pytorch.org/get-started/locally/) make your selection according to your device, so you get your suitable command at bottom: "Run this Command"
run that command in CMD
